yesod-core/Yesod/Routes/Parse.hs

```diff
@@ -102,6 +102,26 @@ resourcesFromString =
                 [] -> (id, otherLines)
                 _ -> error $ "Invalid resource line: " ++ thisLine
 
+-- | Splits a string by spaces, as long as the spaces are not enclosed by curly brackets (not recursive).
+splitSpaces :: String -> [String]
+splitSpaces "" = []
+splitSpaces str = 
+    let (rest, piece) = parse $ dropWhile isSpace str in
+    piece:(splitSpaces rest)
+
+    where 
+        parse :: String -> ( String, String)
+        parse ('{':s) = fmap ('{':) $ parseBracket s
```

Comment:
I think that would be too permissive. Otherwise, lines could contain arbitrary braces in routes. For example, the following would probably parse:

    /m{}y/{route with}/{spaces} RouteR GET

I prefer the current stricter grammar that requires the braces follow `'#'` so that braces can only surround types in the route.